app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.maps.testapp.annotation
+
+import android.graphics.Color
+import android.os.Handler
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import com.mapbox.geojson.Point
+import com.mapbox.maps.RenderMode
+import com.mapbox.maps.extension.style.utils.ColorUtils
+import com.mapbox.maps.plugin.annotation.generated.Symbol
+import com.mapbox.maps.plugin.annotation.generated.SymbolManager
+import com.mapbox.maps.plugin.annotation.generated.SymbolOptions
+import com.mapbox.maps.plugin.annotation.generated.getSymbolManager
+import com.mapbox.maps.plugin.annotation.getAnnotationPlugin
+import com.mapbox.maps.plugin.delegates.listeners.OnDidFinishRenderingMapListener
+import com.mapbox.maps.testapp.BaseMapTest
+import com.mapbox.maps.testapp.examples.annotation.AnnotationUtils
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+/**
+ * Instrumented test for verifying annotation updating while changing style
+ */
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class UpdateAnnotationTest : BaseMapTest(), OnDidFinishRenderingMapListener {
+  private lateinit var runnable: Runnable
+  private var index = 0
+  private val latch = CountDownLatch(AnnotationUtils.STYLES.size * 3)
+  private lateinit var symbolManager: SymbolManager
+  private lateinit var symbol: Symbol
+
+  @Test
+  fun testUpdateAnnotation() {
+    mapboxMap.addOnDidFinishRenderingMapListener(this)
+
+    rule.scenario.onActivity {
+      val handler = Handler(it.mainLooper)
+
+      it.runOnUiThread {
+        mapboxMap.loadStyleUri(AnnotationUtils.STYLES[index++ % AnnotationUtils.STYLES.size]) {
+          symbolManager = mapView.getAnnotationPlugin().getSymbolManager(mapView)
+          symbol = symbolManager.create(
+            SymbolOptions()
+              .withIconColor(ColorUtils.colorToRgbaString(Color.RED))
```

Comment:
can we add maximum possible options here?

Reply:
The purpose of this test is to verify no crash happen while changing style and update at the same time. That's said, the inside source and layer are handled properly, which has been done in https://github.com/mapbox/mapbox-maps-android/pull/47/files#diff-4439957e75cc10a9ecf0041db71889e34906978146ed8b09519c8e0cc7405a1fR142. No matter how many options are added here actually has no effect for it.

Reply:
But if some option could not get applied and causes a crash - we could catch it here, couldn't we?
Leaving the decision up to you here.

Reply:
🤔 , Let's add some more options to cover a wider scope.